tests/v2/utils/library/project_test_utils.go

```diff
@@ -0,0 +1,192 @@
+package utils
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/google/go-cmp/cmp"
+	"sigs.k8s.io/yaml"
+
+	schema "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+	commonUtils "github.com/devfile/api/v2/test/v200/utils/common"
+)
+
+// getSchemaProject gets a named Project from the saved devfile schema structure
+func getSchemaProject(projects []schema.Project, name string) (*schema.Project, bool) {
+	found := false
+	var schemaProject schema.Project
+	for _, project := range projects {
+		if project.Name == name {
+			schemaProject = project
+			found = true
+			break
+		}
+	}
+	return &schemaProject, found
+}
+
+// getSchemaStarterProject gets a named Starter Project from the saved devfile schema structure
+func getSchemaStarterProject(starterProjects []schema.StarterProject, name string) (*schema.StarterProject, bool) {
+	found := false
+	var schemaStarterProject schema.StarterProject
+	for _, starterProject := range starterProjects {
+		if starterProject.Name == name {
+			schemaStarterProject = starterProject
+			found = true
+			break
+		}
+	}
+	return &schemaStarterProject, found
+}
+
+// UpdateProject randomly modifies an existing project
+func UpdateProject(devfile *commonUtils.TestDevfile, projectName string) error {
+
+	var err error
+	testProject, found := getSchemaProject(devfile.SchemaDevFile.Projects, projectName)
+	if found {
+		commonUtils.LogInfoMessage(fmt.Sprintf("Updating Project : %s", projectName))
+		devfile.SetProjectValues(testProject)
+	} else {
+		err = errors.New(commonUtils.LogErrorMessage(fmt.Sprintf("Project not found in test : %s", projectName)))
+	}
+	return err
+
+}
+
+// UpdateStarterProject randomly modifies an existing starter project
+func UpdateStarterProject(devfile *commonUtils.TestDevfile, projectName string) error {
+
+	var err error
+	testStarterProject, found := getSchemaStarterProject(devfile.SchemaDevFile.StarterProjects, projectName)
+	if found {
+		commonUtils.LogInfoMessage(fmt.Sprintf("Updating Starter Project : %s", projectName))
+		devfile.SetStarterProjectValues(testStarterProject)
+	} else {
+		err = errors.New(commonUtils.LogErrorMessage(fmt.Sprintf("Starter Project not found in test : %s", projectName)))
+	}
+	return err
+}
+
+// VerifyProjects verifies projects returned by the parser are the same as those saved in the devfile schema
+func VerifyProjects(devfile *commonUtils.TestDevfile, parserProjects []schema.Project) error {
```

Comment:
shouldnt this be called from `verify`

Reply:
Yes, lol, not sure how I lost that code. Too many branches!